src/Parser/Druid/Feral/Modules/Bleeds/RipSnapshot.js

```diff
@@ -0,0 +1,237 @@
+    }
+    if (SPELLS.FEROCIOUS_BITE.id === event.ability.guid) {
+      debug && console.log(`${this.owner.formatTimestamp(event.timestamp)} bite cast.`);
+      this.handleBiteExtend(event);
+    }
+  }
+
+  on_byPlayer_spendresource(event) {
+    if (SPELLS.RIP.id === event.ability.guid &&
+        event.resourceChangeType === RESOURCE_TYPES.COMBO_POINTS.id) {
+      this.comboLastRip = event.resourceChange;
+    }
+  }
+
+  // use damage events to keep track of each target's health to tell when they're in "execute" range
+  on_byPlayer_damage(event) {
+    // no need to track health if combatant has sabertooth, and don't track health of friendlies.
+    if (this.hasSabertooth || event.targetIsFriendly) {
+      return;
+    }
+    this.healthFraction[encodeTargetString(event.targetID, event.targetInstance)] = event.hitPoints / event.maxHitPoints;
+  }
+
+  biteCanRefresh(target) {
+    return this.hasSabertooth || 
+      (this.healthFraction[target] && this.healthFraction[target] < BITE_EXECUTE_RANGE);
+  }
+
+  handleBiteExtend(event) {
+    const target = encodeTargetString(event.targetID, event.targetInstance);
+    if (!this.biteCanRefresh(target)) {
+      return;
+    }
+    const existing = this.stateByTarget[target];
+    if (!existing || existing.expireTime < event.timestamp) {
+      return;
+    }
+
+    // Bite "refreshes the duration of your Rip on the target" which means setting to the base duration.
+    existing.expireTime = event.timestamp + this.durationOfFresh;
+    existing.pandemicTime = event.timestamp + this.durationOfFresh * (1.0 - PANDEMIC_FRACTION);
+    debug && console.log(`${this.owner.formatTimestamp(event.timestamp)} bite extended rip to ${this.owner.formatTimestamp(existing.expireTime)}`);
+  }
+
+  checkRefreshRule(stateNew) {
+    const stateOld = stateNew.prev;
+    if (!stateOld || stateOld.expireTime < stateNew.startTime) {
+      return;
+    }
+    const event = stateNew.castEvent;
+    if (!event) {
+      debug && console.warn('RipSnapshot checking refresh rule with a state that was never assigned a cast event.');
+      return;
+    }
+
+    const target = encodeTargetString(event.targetID, event.targetInstance);
+    if (this.biteCanRefresh(target) &&
+        stateOld.power >= stateNew.power) {
+      this.shouldBeBiteCount += 1;
+      event.meta = event.meta || {};
+      event.meta.isInefficientCast = true;
+      const strengthComment = (stateOld.power > stateNew.power) ? 'a stronger' : 'the same strength';
+      event.meta.inefficientCastReason = `Used Rip when you could have extended using Ferocious Bite and kept ${strengthComment} snapshot.`;
+    }
+    
+    if (stateOld.pandemicTime > stateNew.startTime &&
+        stateOld.power > stateNew.power) {
+      this.downgradeCount += 1;
+
+      // this downgrade is relatively minor, so don't overwrite cast info from elsewhere
+      if (!event.meta || (!event.meta.isInefficientCast && !event.meta.isEnhancedCast)) {
```

Comment:
can check for the opposite and return early here to avoid the nested if statements